### ISSUE TITLE
Couldn't execute tests/run project - x86_64-apple-macos vs arm64-apple-macos issue

### ISSUE THREAD

**Issue by magdaostaszewska** [Tyler-Keith-Thompson/CucumberSwift#69]
**Describe the bug**
After adding CucumberSwift via SPM to our project we were facing with a lot of issues with libraries (our project is very characteristic and has a lot of dependencies) but we only have one left which is related to CucumberSwift:

`/Repos/app/UITests/FirstCucumberTests.swift:11:8 Could not find module 'CucumberSwift' for target 'x86_64-apple-macos'; found: arm64-apple-macos, at: /Library/Developer/Xcode/DerivedData/App-eaadrssknjauggcltgqnqhhvhwzo/Build/Products/Local/CucumberSwift.swiftmodule
`
Path has been changed for security.

**To Reproduce**
1. Setup CucumberSwift via SPM
2. Do everything from https://tyler-keith-thompson.github.io/CucumberSwift/tutorials/cucumberswift/spm-step-by-step 
3. Try run tests/run project etc

**Expected behavior**
Tests should execute, project should run, for sure if should build itself.

**Extra info**
Tried run Xcode as Rosetta - same result
Tried change architecture for not standard one (like you could find on internet) - same result
Tried changing targets/files/tests/features files etc - same result
Tried clean build folder/ Devired data/ SPM etc with restarting Xcode - same result
Tried previous versions of CucumberSwift (from the beginning of December even) - same result

**Additional context**
 - XCode Version: 14.2
 - CucumberSwift Version: main
 - Example Feature File - same like in the instruction

**Comment by Tyler-Keith-Thompson** [comment]
That's odd, I run on an M1 and the library isn't doing anything too crazy.

I'll look into this as soon as I can, but at the moment I've no idea why you're running into trouble.

Does the sample app work for you?

**Comment by Tyler-Keith-Thompson** [comment]
Sorry to suggest something so silly, but I've had at least one other instance where I ran into a similarly obscure error and it turns out I'd linked against the production code target rather than the test target. You might double check that.

Then again I've got another report of problems with Xcode 14.2 and the latest CucumberSwift, so it's entirely possible that we've got a bug right now that my pipelines aren't reproducing.

**Comment by magdaostaszewska** [comment]
Hi there!
I apologise that you were waiting for the answer! 
Like i mentioned - our project is very characteristic and has a lot of dependencies. And that was the reason - we have several architectures inside to build and firstly to even use CucumberSwift we needed to mess up a little there and later, when CucumberSwift was there we just tried to go back to previous settings. And it works.
More precisely, we had to change to `YES` for `.local' environments and change to `NO` for `.production` and `.release` part.
Thank you for your help and really quick response!
